recipes/libxml2/cmake/conanfile.py

```diff
@@ -0,0 +1,151 @@
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.cache_variables["BUILD_SHARED_LIBS"] = self.options.shared
+        tc.cache_variables["LIBXML2_WITH_CATALOG"] = True
+        tc.cache_variables["LIBXML2_WITH_DEBUG"] = True
+        tc.cache_variables["LIBXML2_WITH_HTML"] = True
+        tc.cache_variables["LIBXML2_WITH_HTTP"] = True
+        tc.cache_variables["LIBXML2_WITH_ICONV"] = self.options.iconv
+        tc.cache_variables["LIBXML2_WITH_ICU"] = self.options.icu
+        tc.cache_variables["LIBXML2_WITH_ISO8859X"] = True
+        tc.cache_variables["LIBXML2_WITH_LEGACY"] = False
+        tc.cache_variables["LIBXML2_WITH_LZMA"] = self.options.lzma
+        tc.cache_variables["LIBXML2_WITH_MODULES"] = True
+        tc.cache_variables["LIBXML2_WITH_OUTPUT"] = True
+        tc.cache_variables["LIBXML2_WITH_PATTERN"] = True
+        tc.cache_variables["LIBXML2_WITH_PROGRAMS"] = self.options.programs
+        tc.cache_variables["LIBXML2_WITH_PUSH"] = True
+        tc.cache_variables["LIBXML2_WITH_PYTHON"] = False
+        tc.cache_variables["LIBXML2_WITH_READLINE"] =False
+        tc.cache_variables["LIBXML2_WITH_REGEXPS"] = True
+        tc.cache_variables["LIBXML2_WITH_SAX1"] = True
+        tc.cache_variables["LIBXML2_WITH_THREADS"] = True
+        tc.cache_variables["LIBXML2_WITH_TLS"] = False
+        tc.cache_variables["LIBXML2_WITH_VALID"] = True
+        tc.cache_variables["LIBXML2_WITH_XINCLUDE"] = True
+        tc.cache_variables["LIBXML2_WITH_XPATH"] = True
+        tc.cache_variables["LIBXML2_WITH_ZLIB"] = self.options.zlib
+        tc.cache_variables["LIBXML2_WITH_C14N"] = True
+        tc.cache_variables["LIBXML2_WITH_HISTORY"] = False
+        tc.cache_variables["LIBXML2_WITH_SCHEMAS"] = True
+        tc.cache_variables["LIBXML2_WITH_SCHEMATRON"] = True
+        tc.cache_variables["LIBXML2_WITH_THREAD_ALLOC"] = False
+        tc.cache_variables["LIBXML2_WITH_WRITER"] = True
+        tc.cache_variables["LIBXML2_WITH_XPTR"] = True
+        tc.cache_variables["LIBXML2_WITH_RELAXNG"] = True
+
+        # inhibit any CMake logic that requires pkg-config
+        tc.cache_variables["PKG_CONFIG_EXECUTABLE"] = "PKG_CONFIG_EXECUTABLE-NOTFOUND"
+        tc.cache_variables["LIBXML2_WITH_TESTS"] = False
+        tc.generate()
+
+        cmake_deps = CMakeDeps(self)
+        cmake_deps.generate()
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def package(self):
+        cmake = CMake(self)
+        cmake.install()
+
+        copy(self, "Copyright", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
+
+        rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))
+        rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
+        rmdir(self, os.path.join(self.package_folder, "share")) # contains manpages and docs - if needed, please open an issue
+
+    def package_info(self):
```

Comment:
There's a missing libiconv require when enabled

`ERROR: libxml2/2.14.5: package_info(): The direct dependency 'libiconv' is not used by any '(cpp_info/components).requires'.`

Reply:
Thanks! Only happens when icu is enabled - fixed!